app/lib/report_helper.rb

```diff
@@ -103,53 +119,55 @@ def taxonomy_details(report, params)
-      category_info = cat_name_info[category_taxid]
-      category_name = category_info ? category_info[0]['name'] : 'Other'
-
-      taxon_nts = tax_hit[[tax_id, 'NT']] || [nil]
-      taxon_nrs = tax_hit[[tax_id, 'NR']] || [nil]
-
-      if level[tax_id] == species_level
-        species_nts = taxon_nts
-        species_nrs = taxon_nrs
-        genus_taxid = linfo["genus_taxid"]
-        genus_nts = tax_hit[[genus_taxid, 'NT']] || [nil]
-        genus_nrs = tax_hit[[genus_taxid, 'NR']] || [nil]
-      else
-        # assert level[tax_id] == genus_level
-        species_nts = [nil]
-        species_nrs = [nil]
-        genus_nts = taxon_nts
-        genus_nrs = taxon_nrs
-      end
-
-      details = {
-        nt_ele: species_nts[0],
-        nr_ele: species_nrs[0],
-        category: category_name,
-        genus_nt_ele: genus_nts[0],
-        genus_nr_ele: genus_nrs[0]
-      }
+      if level[tax_id] == view_level_int
+        linfo = lineage_info[tax_id] || [{}]
+        linfo = linfo.first
+
+        category_taxid = linfo["superkingdom_taxid"]
+        category_info = cat_name_info[category_taxid]
+        category_name = category_info ? category_info[0]['name'] : 'Other'
+
+        taxon_nts = tax_hit[[tax_id, 'NT']] || [nil]
+        taxon_nrs = tax_hit[[tax_id, 'NR']] || [nil]
+
+        if level[tax_id] == species_level
+          species_nts = taxon_nts
+          species_nrs = taxon_nrs
+          genus_taxid = linfo["genus_taxid"]
+          genus_nts = tax_hit[[genus_taxid, 'NT']] || [nil]
+          genus_nrs = tax_hit[[genus_taxid, 'NR']] || [nil]
+        else
+          # assert level[tax_id] == genus_level
+          species_nts = [nil]
+          species_nrs = [nil]
+          genus_nts = taxon_nts
+          genus_nrs = taxon_nrs
+        end
 
-      # For now we sort by species RPM or Z
-      # TODO: Allow sorting by genus RPM or Z
-      sort_details = details[sort_details_key]
-      if sort_details
-        sort_key = sort_details[sort_field]
-        if sort_key
-          sort_key = 0.0 - sort_key if sort_direction == :highest
-          details[:sort_key] = sort_key
+        details = {
+          nt_ele: species_nts[0],
+          nr_ele: species_nrs[0],
+          category: category_name,
+          genus_nt_ele: genus_nts[0],
+          genus_nr_ele: genus_nrs[0]
+        }
+
+        # For now we sort by species RPM or Z
+        # TODO: Allow sorting by genus RPM or Z
+        sort_details = details[sort_details_key]
+        if sort_details
+          sort_key = sort_details[sort_field]
+          if sort_key
+            sort_key = 0.0 - sort_key if sort_direction == :highest
+            details[:sort_key] = sort_key
+          end
         end
-      end
 
-      if details[:sort_key]
-        sortable.push(details)
-      else
-        unsortable.push(details)
+        if details[:sort_key]
+          sortable.push(details)
+        else
+          unsortable.push(details)
+        end
```

Comment:
Please move the filtering code from `taxon_zscores.keep_if` to down here.   We don't want the taxon_zscores to be filtered before this loop, because that would potentially throw away some of the genus taxons before they are accessed from species taxons here.

Reply:
Good point. I will leave as is so you can fix it in your upcoming PR as discussed.